controllers/controllers/workloads/cfapp_controller.go

```diff
@@ -283,3 +290,54 @@ func (r *CFAppReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&korifiv1alpha1.CFApp{}).
 		Complete(r)
 }
+
+func (r *CFAppReconciler) CreateVCAPServicesSecretForApp(ctx context.Context, cfApp *korifiv1alpha1.CFApp) error {
+	if cfApp.Status.VCAPServicesSecretName != "" {
+		return nil
+	}
+
+	vcapServicesSecretName := cfApp.Name + "-vcap-services"
+	vcapServicesSecretLookupKey := types.NamespacedName{Name: vcapServicesSecretName, Namespace: cfApp.Namespace}
+	vcapServicesSecret := new(corev1.Secret)
+	err := r.Client.Get(ctx, vcapServicesSecretLookupKey, vcapServicesSecret)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			r.Log.Error(err, "unable to fetch vcap services Secret")
+			return err
+		}
+
+		vcapServicesSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      vcapServicesSecretName,
+				Namespace: cfApp.Namespace,
+			},
+			Immutable: nil,
+			Data:      nil,
+			StringData: map[string]string{
+				"VCAP_SERVICES": "{}",
+			},
+			Type: "",
+		}
+		err = r.Client.Create(ctx, vcapServicesSecret)
+		if err != nil {
+			r.Log.Error(err, "unable to create vcap services Secret")
+			return err
+		}
+	}
+	originalCFApp := cfApp.DeepCopy()
+
+	cfApp.Status.VCAPServicesSecretName = vcapServicesSecretName
+
+	if cfApp.Status.ObservedDesiredState != cfApp.Spec.DesiredState {
+		cfApp.Status.ObservedDesiredState = cfApp.Spec.DesiredState
+	}
+	if cfApp.Status.Conditions == nil {
```

Comment:
I don't think that `cfApp.Status.Condition` can ever be nil in a `Reconcile` context. I remember trying to update `Status` with a `nil` `Condition` caused some k8s validation to fail

Reply:
I'm not 100% sure on this one, but I thought we saw this behaviour during a test. Not sure if it was an integration or e2e test though.